source/scene.ts

```diff
@@ -118,47 +95,53 @@ function init() {
     const plane = new Mesh(planeGeometry, planeMaterial)
     plane.rotateX(Math.PI / 2)
     plane.receiveShadow = true
-
-    scene.add(cube)
     scene.add(plane)
+
+    // cubes
+    const gridSize = 10;
+    const spacing = 1.2;
+    const gridCenterOffset = (gridSize - 1) * spacing * 0.5;
+
+    const sideLength = 1;
+    const minHeight = 0.2;
+    const maxHeight = 2.0;
+
+    cubeGroup = new Group();
+    const cubeGeometry = new BoxGeometry(sideLength, sideLength, sideLength);
+
+    for (let i = 0; i < gridSize; i++) {
+      for (let j = 0; j < gridSize; j++) {
+        
+        const randomHeight = Math.random() * (maxHeight - minHeight) + minHeight;
+        const randomColor = new Color('#ff66a1').lerp(new Color('#ffffff'), Math.random());
+
+        const cubeMaterial = new MeshStandardMaterial({
+          color: randomColor,
+          metalness: 0.5,
+          roughness: 0.7,
+        });
+
+        const tmpCube = new Mesh(cubeGeometry, cubeMaterial);
+        tmpCube.castShadow = true;
+
+        tmpCube.position.set(i * spacing - gridCenterOffset, randomHeight / 2, j * spacing - gridCenterOffset);
```

Comment:
Maybe add small constant (0.01) to y-coordinate to prevent z-fighting with plane when viewed from below